src/psyclone/tests/domain/lfric/lfric_loop_test.py

```diff
@@ -369,21 +404,25 @@ def test_loop_stop_expr_intergrid(dist_mem):
     loops = sched.walk(DynLoop)
     ubound = loops[1].stop_expr
     assert isinstance(ubound, ArrayReference)
-    assert ubound.symbol.name == "last_cell_all_colours_field1"
     assert ubound.indices[0].name == "colour"
     if dist_mem:
+        assert ubound.symbol.name == "last_halo_cell_all_colours_field1"
         assert isinstance(ubound.indices[1], Literal)
         assert ubound.indices[1].value == "1"
         # Alter the loop so that it goes to the full halo depth
         loops[1]._upper_bound_halo_depth = None
         ubound = loops[1].stop_expr
         assert isinstance(ubound.indices[1], Reference)
         assert ubound.indices[1].symbol.name == "max_halo_depth_mesh_field1"
+    else:
+        assert ubound.symbol.name == "last_edge_cell_all_colours_field1"
 
 
 def test_lfricloop_gen_code_err():
     ''' Test that the gen_code method raises the expected exception if the loop
-    type is 'colours' and is within an OpenMP parallel region. '''
+    type is 'colours' and is within an OpenMP parallel region.
```

Comment:
I would suggest single quotes around `'gen_code'` in the line above.

Reply:
done